% CHAPTER 2
\chapter{Material and Method}~\label{chp:b2}
\section{Datasets}
In this study, I analyzed 19 microarray gene expression datasets to investigate age-related gene expression heterogeneity change in human brain during development and aging.
The datasets were retrieved from 3 independent published studies, containing microarray data for the human brain~\cite{Colantuoni2011, Kang2011, Somel2011, Somel2010}.
Overall, the datasets include 1010 samples from 298 individuals spanning 17 different brain region, which are not mutually exclusive.
All datasets have samples covering whole lifespan with ages ranging from 0 to 98 years (\textbf{Figure~\ref{fig:fig2.1}}).
A summary of datasets used in this study is shown in \textbf{Table~\ref{table:table1}}.

It should also be noted that Kang2011 datasets contain samples from left and right hemispheres of the same individual.
These samples were analyzed as biological replicates, meaning that samples were not seperated into two different datasets, for three reasons.
First, it was previously suggested that left and right hemispheres of the brain may show asymmetric age-related changes~\cite{Sun2005, Dolcos2002}.
Second, the other datasets do not contain hemisphere information.
Last, previous studies analyzing this dataset, including the original study, also treated them as biological replicates~\cite{Kang2011,Donertas2017}.

Additionally, Somel2011\char`_PFC dataset includes two pairs of technical replicates, between which the correlation was high. 
Therefore, the mean of expression values was used in the downstream analysis.


The datasets were downloaded from NCBI Gene Expression Omnibus (GEO) database using the accession codes given in the
\textbf{Table~\ref{table:table1}}. 
All the analysis was performed in R programming environment.

\begin{figure}[h]
\centering
\includegraphics[width=.9\textwidth]{figures/figure2_1.png}
\caption{The distribution of ages of the samples. (a) Number of samples included in age intervals. (b) Distribution of ages. The color coding reflect different data sources. }\label{fig:fig2.1}
\end{figure}

\begin{table}[ht]
\centering
\caption{The list of microarray human brain gene expression datasets. The sample sizes were calculated after the removal of outliers.}\label{table:table1}
%% \resizebox{\textwidth}{!}{\begin{tabular}{||c c c c||} 
\begin{tabular}{|c c c c|}
 \hline
 \textbf{GEO Acc.} & \textbf{Source} & \textbf{Brain Region} & \textbf{Sample Size} \\ [0.5ex] 
 \hline\hline
 GSE30272 & Colantuoni2011 & PFC & 231 \\ 
 \hline
 GSE25219 & Kang2011 & A1C & 47 \\
 \hline
 GSE25219 & Kang2011 & AMY & 43 \\
 \hline
 GSE25219 & Kang2011 & CBC & 47 \\
 \hline
 GSE25219 & Kang2011 & DFC & 48 \\
 \hline
 GSE25219 & Kang2011 & HIP & 39 \\
 \hline
 GSE25219 & Kang2011 & IPC & 49 \\
 \hline
 GSE25219 & Kang2011 & ITC & 49 \\
 \hline
 GSE25219 & Kang2011 & M1C & 45 \\
 \hline
 GSE25219 & Kang2011 & MD & 43 \\
 \hline
 GSE25219 & Kang2011 & MFC & 50 \\
 \hline
 GSE25219 & Kang2011 & OFC & 48 \\
 \hline
 GSE25219 & Kang2011 & S1C & 46 \\
 \hline
 GSE25219 & Kang2011 & STC & 48 \\
 \hline
 GSE25219 & Kang2011 & STR & 41 \\
 \hline
 GSE25219 & Kang2011 & V1C & 48 \\
 \hline
 GSE25219 & Kang2011 & VFC & 47 \\
 \hline
 GSE22569 & Somel2011 & PFC & 23 \\
 \hline
 GSE18069 & Somel2011 & CBC & 22 \\
\hline
\end{tabular}
\end{table}

\subsection{Dataset selection}
The age-series datasets analyzed in this study are all microarray-based. 
Although there was one other RNA-Sequencing based dataset that covers whole lifespan~\cite{Mazin2013}, I chose not to include it in this anaysis for two reasons.
First, the samples were already included in the Somel2011 dataset. 
Second, it is an underpowered dataset with data from only 35 individuals that cannot reliably lead to conclusion.

There were also RNA-Sequencing datasets containing samples from only development or aging periods. 
Since combining independent development and aging datasets may confound biological effects that I aimed to examine, 
this study was limited to meta-analysis of 19 microarray-based datasets. 

\subsection{Seperating development and aging datasets}
The aim of this study to investigate age-related gene expression change during development and aging. 
Thus, all the datasets were seperated into two datasets: development (0 to 20 years of age) and aging (20 to 98 years of age).
The age of 20 years was used to seperate development and aging for the following reasons:
\begin{enumerate}
    \item The age of 20 was shown to correspond approximately to the age of reproduction in human societies~\cite{Walker2006}.
    \item Previous studies investigating age-related gene expression trajectories demonstrated that 20 years of age corresponds to a turning point of gene expression patterns\cite{Colantuoni2011, Donertas2017,Somel2010}.
    \item Earlier research connected the structural changes occuring in the human brain after the age of 20 to age-related phenotypes~\cite{Sowell2004}.
\end{enumerate}

As a result, I obtained: 
(i) one development and one aging dataset for the Colantuoni2011; 
(ii) 16 development and 16 aging datasets for the Kang2011; and
(iii) two development and two aging datasets for the Somel2011.
Overall, both development and aging datasets resulted in a comparable number of samples ($n_{development} = 441$; $n_{aging}=569$).

Moreover, it is important to note that I excluded samples from prenatal development period, 
since gene expression trajectories were shown to be discontinuous between prenatal and postnatal development period~\cite{Colantuoni2011,Kang2011}, 
and since the scope of this study is limited to investigate changes in gene expression heterogeneity during aging compared to pre-adulthood.

\section{Dataset Preprocessing}
Microarray technology is a widely used tool to quantify expression level of gene transcripts from a given sample. 
A microarray chip contains known sequences of oligonucleotides -known as probes- that are located on a solid surface.
Typically, each transcript is represented by a set of 11-20 pairs of probe, called as the probe-set of that transcript, in Affymetrix microarray platforms.
The cDNAs derived from the mRNA transcripts of the sample are hybridised to target probes labelled by detectable fluorochrome molecules, 
where the amount of hybridization is reflected by the light intensity levels.
The quantification of expression is then performed by measuring light intensity levels of each probe, which are stored in CEL files.

The Kang2011 and Somel2011 datasets were generated by Affymetrix HuEx-1\char`_0-st and HuGene-1\char`_0-st microarray platforms, respectively. 
Colantuoni2011 dataset, on the other hand, was generated using HEEBO-7 set (Human 49K oligo array), which is an Illumina based array. 
Since there is no public R library available to process Illumina based data from Colantuoni2011, 
I used the expression data preprocessed by the authors of the original study~\cite{Colantuoni2011}. 
For the datasets from Kang2011 and Somel2011 sources, I downloaded CEL files from GEO database~\cite{Barrett2013}. 
The preprocessing of Kang2011 and Somel2011 datasets can be summarized in four steps: (1) RMA convolution, (2) probe-set summarization,
(3) log2 transformation, and (4) quantile normalization. 
Distribution of expression values after each preprocessing step is shown in \textbf{Figure~\ref{fig:fig2.1}}.
For the Colantuoni2011 dataset, quantile normalization was performed on the preprocessed data.

\subsection{RMA correction}
The very first step of a microarray analysis is the removal of noise and biases from the raw data obtained from light intensities.
There can be a number of factors contributing background errors, such as optical noise, unspecific hybridization and incomplete washing~\cite{Bengtsson2006}. 
Nevertheless, low-level preprocessing and normalization, having a significant effect on the downstream analysis, 
were suggested to be one of the most important step in any microarray data analysis~\cite{Bengtsson2006}.

In this study, background normalization was performed by the Robust Multiarray Average (RMA) convolution method, 
which is a one of the most commonly used method to perform background normalization on microarray data. 
The RMA method involves the removal of technical artefacts so that the measurements from neighbouring probes do not interfere with each other~\cite{Irizarry2003}.

Apart from background normalization, the RMA algorithm also performs probe to probe-set summarization. 
Since each transcript is represented by a set of 11-20 probes, it is necessary to summarize probe-level data into probe-sets, 
by grouping probes corresponding the same transcript. I used the R ``oligo'' library to perform RMA correction~\cite{Carvalho2010}.
As previously stated, RMA correction was performed only on Kang2011 and Somel2011 datasets.

\subsection{Probe-set summarization}
I next summarized the probe-set expression values into gene expression values. 
This step is required to combine and analyze data from different platforms, since we need to have expression values that are defined universally (i.e., by gene IDs).
Thus, converting probe-sets into gene IDs allows us to compare expression levels of the genes among different platforms.

However, probe-set to gene ID conversion is not always one-to-one in many platforms. 
There can be multiple probe-sets that correspond to the same gene, while it is also possible to have a probe-set that maps to multiple genes. 
In this study, probe-sets that correspond to more than one genes were removed, since keeping these samples may lead to a pseudoreplication problem, 
where the expression level of multiple genes would not be independent. 
For the genes having multiple probe-set data, the expression values were calculated by taking the average of the expression values of the probe-sets corresponding the gene.

For the dataset generated by HuGene-1\char`_0-st (Somel2011 datasets), 
Ensembl v.84 annotations~\cite{Yates2016} were retrieved through ``biomaRt'' library in R~\cite{Durinck2009}.
For the Kang2011 dataset generated by HuEx-1\char`_0-st platform, the GPL file deposited in the GEO database was used since the IDs of probe-sets were not complete in Ensembl.
Lastly, for the Colantuoni2011 dataset, the gene IDs were retrieved from the GPL file deposited in the GEO database.

\subsection{Log2 transformation}

The $Log_2$ transformation is the most widely used transformation in microarray data analysis
to remove the correlation between mean and variance, and to make the variance more comparable.
Moreover, in the data produced by microarray platforms, there are typically many genes having lower expression values, 
whereas there are fewer genes having high expression levels,
leading to a right-skewed distribution (\textbf{Figure~\ref{fig:fig2.2}}).
Making distributions more similar, $Log_2$ transformation allows us to perform parametric statistical tests, as most of them assume equal variance. 
Additionally, $Log_2$ transformation also allows us to visualize the data more easily.

Therefore, $Log_2$ transformation was applieid to the expression data from Kang2011 and Somel2011 sources.

\subsection{Quantile normalization}
The microarray platforms are susceptible to technical variation from different sources, hampering the meta analysis of multiple datasets from different sources.
Quantile normalization is one of commonly used methods used to minimize technical variation~\cite{Zhao2020}. 

Quantile normalization assumes the same distribution for all samples. 
Therefore, any significant variation in the distribution shape are regarded as unwanted and non-biological noise, and are eleminated.
However, it is also important to note that quantile normalization should be used with caution as it may also remove signals that can be of biological interest~\cite{Hicks2014}.
Despite this danger, the quantile normalization was performed on all datasets for three reasons. 
First, all the datasets analyzed in this study contain samples only from brain (i.e., same tissue).
Second, sample collection was performed under similar conditions (i.e., from healthy individuals), combined with the first reason, they indicate that overall expression distributions should be similar.
Third, this study mainly focuses on consistent patterns among different datasets, rather than focusing on significant changes of individual genes from a single dataset.
We expect quantile normalization to eleminate only random confounding factors that are not shared among datasets. 
Then, the shared patterns among different datasets were considered as a potential biological signal.

The ``preprocessCore'' R package was used to perform quantile normalization~\cite{Bolstad2021}.

\begin{figure}[h]
\centering
\includegraphics[width=.9\textwidth]{figures/figure2_2.png}
\caption{Summary of preprocessing steps for Somel2011\char`_CBC dataset at different preprocessing steps. 
The top right histogram shows the distribution of raw probe expression levels. 
The top left histogram shows probe-set expression values after RMA correction. 
The bottom left plot shows the gene expression values after RMA correction and Log2 transformation.
The bottom right plot shows the the gene expression values after RMA correction, Log2 transformation and quantile normalization.}~\label{fig:fig2.2}
\end{figure}

\subsection{Scaling}
Next, the expression levels for each gene for each dataset were scaled to $mean=0$ and $standard\ deviation=1$. 
Since linear regression analysis was performed in the downstream analysis, 
it is important to scale the genes before model fitting in order to obtain comparable residuals.
Scaling was performed by using the following formula.
\begin{equation}
    Scaled\ Expr_i = \frac{Expr_i - mean(Expr)}{Standard\ Deviation(Expr)}
    \label{eq:scaling}
\end{equation}
where $Expr_i$ and $Scaled\ Expr_i$ represent expression and scaled expression value of the sample $i$, respectively, 
whereas $Expr$ is the expression values of the gene for all samples in a dataset.

I used ``scale'' function in base R to scale each gene in each dataset seperately.

\subsection{Batch-effect correction}
There was one additional normalization step applied to Somel2011\char`_PFC dataset to correct batch effect.
The correction was performed in three steps as follows:

\begin{enumerate}
    \item For each probe-set, mean expression values were calculated.
    \item Each batch was scaled separately to $mean=0$ and $standard\ deviation=1$ using the equation~\ref{eq:scaling}.
    \item The mean values calculated at step 1 were added to each value.
\end{enumerate}



\section{Outlier removal}
In addition to technical variance introduced by microarray analysis, 
there can be some samples showing greater divergence from the rest of the samples due to unknown factors,
such as different diease backgrounds or different diets.
Including such samples (i.e., outliers) in the analysis would introduce extra noise and would obstract to identify the true relationship between age and expression levels.

To identify outliers, Principal Component Analysis (PCA) was used. 
PCA is a dimension reduction method that allows us to visualize high dimensional data and to detect outliers.
The first two principal components, which are the most important components explaning the largest variance, were used to visualize data and to identify outliers.

Consistent with the previous studies~\cite{Donertas2017,Donertas2018}, the following 7 samples were removed from the analysis: 
\begin{enumerate}
    \item 3 years old \textit{GSM705108} from Kang2011 dataset (A1C brain region);
    \item 37 years old \textit{GSM704438} from Kang2011 dataset (CBC brain region);
    \item 42 years old \textit{GSM705202} from Kang2011 dataset (CBC brain region);
    \item 0 year old \textit{GSM704567} from Kang2011 dataset (HIP brain region);
    \item 40 year old \textit{GSM704627} from Kang2011 dataset (HIP brain region);
    \item 70 year old \textit{GSM704226} from Kang2011 dataset (HIP brain region);
    \item 70 year old \textit{GSM704227} from Kang2011 dataset (HIP brain region).
\end{enumerate}

After removing outliers, we obtained 1,010 samples from 298 individuals (\textbf{Table~\ref{table:table1}}).
Lastly, the common genes among all datasets were selected for the downstream analysis ($n=11,137$), 
and the genes for which we don't have a measurement in at least one dataset were removed.

\section{Age-related expression change}
Having preprocessed datasets for both development and aging periods, I next sought to characterize age-related gene expression changes. 
Linear regression was used to quantify the relationship between expression levels and age. 
The following linear model was fitted to each gene for each time period seperately.

\begin{equation}
    Expr_i = \beta_{i0} + \beta_{i1} * Age^{1/4} + \epsilon_i
    \label{eq:exp_change}
\end{equation}

where, $Expr_i$ is the scaled expression value of the i\textsuperscript{th} gene, $\beta_{i0}$ is the intercept, $\beta_{i1}$ is the slope, $Age$ is the age in days, and $\epsilon_i$ is the residual.
The $\beta_{i1}$ values were considered as the measure of age-related expression change. Throughout this study, $\beta_{i1}$ values will be referred as simply `beta' values.
Since it is the slope of the linear model, a negative beta value indicates a decrease in gene expression with age, while positive values indicate an age-related increase in the expression of the corresponding gene. 
The left panels of \textbf{Figure~\ref{fig:fig2.3}} show an example of quantification of age-related expression changes for two genes.

It is also worth noting that we transformed the individual ages into fourth root scale to obtain an approximately uniform distribution of ages across the lifespan.
To test the effect of the age scale, I performed the same analysis using different age scales and confirmed that they also resulted in quantitativelly similar results.

The linear regression was performed by using ``lm'' function in base R for each gene in each dataset (development and aging datasets separately).

\subsection{Multiple testing correction}


\section{Age-related heterogeneity change}

\section{Principal component analysis}


\begin{figure}[h]
\centering
\includegraphics[width=.9\textwidth]{figures/figure2_3.png}
\caption{Summary of the method used to calculate age-related expression change (left panels) and age-related expression heterogeneity change (right panels) during development (a) and aging (b). Adapted from (Isildak et al., 2020)}\label{fig:fig2.3}
\end{figure}

\section{Permutation test}

\subsection{Expected heterogeneity consistency}

\section{Functional analysis}
asfgafs

\section{Effect of sex-specific gene expression}



\begin{equation}
{\bf y}_k = {\bf x}_k * {\bf P}_k
\end{equation}

























