% CHAPTER 3
\chapter{Results}~\label{chp:b3}

In this study, I investigated the age-related gene expression heterogeneity change by analyzing 19 microarray datasets
containing 1,010 samples from diverse brain regions of 298 individuals, and covering whole lifespan.
To compare heterogeneity change occuring during before and after adulthood, each dataset were first seperated into two datasets: development dataset 
(including samples with ages ranging from 0 to 20 years old) and aging dataset (including samples with ages ranging from 20 to 98 years old).
Only the common genes (n = 11,137) were used in the downstream analysis.

\section{Age-related gene expression change}
While the aim of this study to investigate age-related changes in gene expression heterogeneity, I first sought to characterize age-related change in gene expression.
Linear regression was performed to characterize the relationship between gene expression and age for each gene and dataset separately (\textbf{Sectio~\ref{sub:exp-change}} ).
The beta values (i.e., $\beta_{i1}$ from \textbf{Equation~\ref{eq:exp_change}}) obtained from linear regression were considered as a measure of age-related expression change.

\begin{figure}[h]
    \centering
    \includegraphics[width=.9\textwidth]{figures/figure3_1.png}
    \caption{Age-related gene expression change. 
    (a) The heatmap shows the Spearman's correlation coefficient values for each pair of datasets.
    The row and column annotations reflect the brain region, data source and time period of the corresponding dataset.
    The datasets of each period were clustered separately.
    (b) Principal component analysis of age-related expression changes. Each point shows a dataset, 
    where color-coding reflects time period and point-shape shows the data source. 
    The values in paranthesis (on the axis labels) reflect the explained variance by the corresponding dataset.
    The values shown on the plot show the median euclidian distance among development and aging datasets, 
    calculated using PC1 and PC2.
    (c) Barplots show the number of genes showing significant age-related expression change (FDR p-value < 0.05) during development (left side) and aging (right side).
    Color-coding reflects the direction of expression change, which is determined by the sign of the beta value.
    Adapted from Isildak et al., 2020
    }~\label{fig:fig3.1}
\end{figure}

To investigate the coordination in age-related expression change betweeen datasets, 
I calculated pairwise Spearman's correlation test for each pair of datasets among the beta value.
The heatmap in the \textbf{Figure~\ref{fig:fig3.1}a} shows the Spearman's correlation coefficients ($\rho$) for each pair of datasets.
The significance of correlation among datasets were tested by performing permutation test (see \textbf{Section~\ref{sec:perm}}).
Overall, I found that both development (permutation test p-value<0.001, median $\rho$=0.56) and aging (permutation test p-value=0.003, median $\rho$=0.43)
datasets display a modest correlation with the datasets from the same time period, 
whereas the difference between the correlation of development and aging datasets was not significant (permutation test p-value=0.1).
Still, the weaker correlation among aging datasets compared to development may indicate the stochasticity of aging period.

I next performed principal component analysis (PCA), which is a dimension reduction algorithm that is widely used in transcriptome data analysis. 
The beta values obtained for each gene for each dataset (development and aging datasets, seperately) was used to calculate principal components.
The first two principal components (PC1 and PC2), which also explain the most variance, was used to visualize the data.
\textbf{Figure~\ref{fig:fig3.1}b} shows the development and aging datasets projected on PC1 and PC2 axis, where
the first principal component separates development and aging datasets, suggesting that expression changes may display different patterns in development and aging.
By calculating median euclidian distance, I found that the median distance among development datasets is 21, 
whereas the median distance among aging datasets is 77. 
The fact that development datasets are clustered more closely compared to agig datasets may reflect an increase in heterogeneity during agig period.

Then, the number of genes showing significant age-related gene expression change was calculated for development and aging datasets (\textbf{Figure~\ref{fig:fig3.1}c}).
The p-values obtained from linear regression analysis for each gene in each dataset were adjust for multiple testing using the B{\&}H method (see \textbf{Section~\ref{subsec:p.adjust}}).
Analyzing significant changes, I first found that there are significantly more genes in the development datasets showing significant change compared to aging datasets (permutation test p-value = 0.003),
which may again reflect the higher heterogeneity during aging compared to development.
Second, the direction of change in development datasets is mainly in positive direction (14 of 19 datasets showed more increase), 
whereas genes showing significant change during aging tend to decrease in expression (13 of 19 datasets showed more decrease).

Combined, my analysis of age-related expression changes demonstrated that the age-related gene expression changes may be less coordinated during aging,
leading to a higher inter-individual variability during aging, which may be the first clues of increased age-related heterogeneity during aging.

\section{Age-related expression heterogeneity change}
I next characterized age-related change in gene expression heterogeneity by performing Spearman's correlation test between absolute value of residuals and fourth root of age (see \textbf{Section~\ref{sec:het-change}}).
Similar to the previous section, I first examined the correlations in gene expression heterogeneity change among development and aging datasets (\textbf{Figure~\ref{fig:fig3.2}a})
By conducting  Spearman's correlation test between all possible pairs of datasets among rho values, 
I found that aging datasets show higher correlation (median $\rho$=0.21, permutation test p-value=0.24) in heterogeneity change
compared to development datasets (median $\rho$=0.11, permutation test p-value=0.25), although both were not significant.
Although the difference in correlation of heterogeneity change between development and aging datasets were not significant (permutation test p-value=0.2),
this observation may suggest that aging datasets show more similar changes in heterogeneity compared to development datasets.

\begin{figure}[h]
    \centering
    \includegraphics[width=.9\textwidth]{figures/figure3_2.png}
    \caption{Age-related heterogeneity change. 
    (a) The heatmap shows the Spearman's correlation coefficient values for each pair of datasets.
    The row and column annotations reflect the brain region, data source and time period of the corresponding dataset.
    The datasets of each period were clustered separately.
    (b) Principal component analysis of age-related heterogeneity changes. Each point shows a dataset, 
    where color-coding reflects time period and point-shape shows the data source. 
    The values in paranthesis (on the axis labels) reflect the explained variance by the corresponding dataset.
    The values shown on the plot show the median euclidian distance among development and aging datasets, 
    calculated using PC1 and PC2.
    (c) Barplots show the number of genes showing significant age-related heterogeneity change (FDR p-value < 0.05) during development (left side) and aging (right side).
    Color-coding reflects the direction of heterogeneity change, which is determined by the sign of the rho value.
    Adapted from Isildak et al., 2020
    }~\label{fig:fig3.2}
\end{figure}

Principal component analysis of heterogeneity changes (i.e., rho values) was performed using a similar procedure as the previous section, 
except rho values were used instead of beta values. 
Similar to the PCA of expression changes, PCA of heterogeneity changes showed that development and aging datasets can be differentiated from each other by heterogeneity change (\textbf{Figure~\ref{fig:fig3.2}b}).
However, unlike expression change, development datasets tended to cluster more sparsly (median euclidian distance = 44), compared to aging datasets (median euclidian distance = 41).

I then investigated the genes showing significant changes in expression heterogeneity. 
The p-values obtained from Spearman's correlation test for each gene for each dataset were corrected for multiple testing using B{\&}H method (see \textbf{Section~\ref{sec:het-change}}).
(\textbf{Figure~\ref{fig:fig3.2}c}) shows the number of genes showing significant hetergeneity change during development and aging.
First, I found that there is a small number of genes showing significant change during development compared to aging (permutation test p-value < 0.05).
Moreover, genes showing significant heterogeneity change during development were tend to decrease in heterogeneity, whereas they showed a tendency to increase in aging datasets.
It is also important to note the high number of genes showing significant change in Colantuoni2011, which is a result of large sample size leading to an increased statistical power.

Overall, this analysis demonstrated that there may be a higher consistency in heterogeneity change during aging compared to development, 
suggesting that increased heterogeneity may be a characteristic of aging period.

\section{Consistent increase in heterogeneity during aging}
Having observed both that heterogeneity changes can differentiate between time periods and that there is a tendency towards more similar changes in heterogeneity during aging,
I next focused on identifying those genes showing consistent change in heterogeneity.
As previously stated, in this study, I chose to focus on genes showing consistent trends among different datasets, rather than focusing on significant changes within individual datasets.
The most important advantage of this approach is that, by using the advantage of having multiple datasets, 
it enables us to detect genes that would otherwise not pass the significance treshold due to low sample size.

I first analyzed the distribution of heterogeneity changes (i.e., $\rho$ values) for each dataset and time period (\textbf{Figure~\ref{fig:fig3.3}a}).

\begin{figure}[h]
    \centering
    \includegraphics[width=.9\textwidth]{figures/figure3_3.png}
    \caption{(a) Boxplots on the left side shows the distribution of rho values, 
    whereas the barplots on the right shows the difference in medians between development and aging datasets.
    (b) Distribution of Spearman's correlation coefficients between expression and heterogeneity change of each dataset. 
    The color-coding reflects data source, whereas 
    (c) Barplots showing expected and observed number of genes showing consistent increase in heterogeneiy among datasets during development (top panel) and aging (bottom panel).
    Adapted from Isildak et al., 2020
    }~\label{fig:fig3.3}
\end{figure}
