% CHAPTER 3
\chapter{Results}~\label{chp:b3}

In this study, I investigated the age-related gene expression heterogeneity change by analyzing 19 microarray datasets
containing 1,010 samples from diverse brain regions of 298 individuals, and covering the whole lifespan.
To compare heterogeneity change occurring before and after adulthood, each dataset was first separated into two datasets: one development dataset 
(including samples with ages ranging from 0 to 20 years old) and one aging dataset (including samples with ages ranging from 20 to 98 years old).
Only the common genes (n = 11,137) were used in the downstream analysis.

\section{Age-related gene expression change}
While this study aims to investigate age-related changes in gene expression heterogeneity, I first sought to characterize the age-related change in gene expression.
Linear regression was performed to characterize the relationship between gene expression and age for each gene and dataset separately (\textbf{Section~\ref{sec:exp-change}}).
The beta values (i.e., $\beta_{i1}$ from \textbf{Equation~\ref{eq:exp_change}}) obtained from linear regression were considered as a measure of age-related expression change.

\begin{figure}[h]
    \centering
    \includegraphics[width=.9\textwidth]{figures/figure3_1.png}
    \caption{Age-related gene expression change. 
    (a) The heatmap shows Spearman's correlation coefficient values for each pair of datasets.
    The row and column annotations reflect the brain region, data source and time period of the corresponding dataset.
    The datasets of each period were clustered separately.
    (b) Principal component analysis of age-related expression changes. Each point shows a dataset, 
    where color-coding reflects time period and point-shape shows the data source. 
    The values in parenthesis (on the axis labels) reflect the explained variance by the corresponding dataset.
    The values shown on the plot show the median euclidian distance among development and aging datasets, 
    calculated using PC1 and PC2.
    (c) Barplots show the number of genes showing significant age-related expression change (FDR p-value < 0.05) during development (left side) and aging (right side).
    Color-coding reflects the direction of expression change, which is determined by the sign of the beta value.
    Adapted from Isildak et al., 2020
    }~\label{fig:fig3.1}
\end{figure}

To investigate the coordination in age-related expression change between datasets, 
I calculated pairwise Spearman's correlation test for each pair of datasets among the beta value.
The heatmap in the \textbf{Figure~\ref{fig:fig3.1}a} shows the Spearman's correlation coefficients ($\rho$) for each pair of datasets.
The significance of correlation among datasets was tested by performing a permutation test (see \textbf{Section~\ref{sec:perm}}).
Overall, I found that both development (permutation test p-value < 0.001, median $\rho$ = 0.56) and aging (permutation test p-value = 0.003, median $\rho$ = 0.43)
datasets display a modest correlation with the datasets from the same time period, 
whereas the difference between the correlation of development and aging datasets was not significant (permutation test p-value = 0.1).
Still, the weaker correlation among aging datasets compared to development may indicate the stochasticity of the aging period.

I next performed principal component analysis (PCA), which is a dimension reduction algorithm that is widely used in transcriptome data analysis. 
The beta values obtained for each gene for each dataset (development and aging datasets, separately) were used to calculate principal components.
The first two principal components (PC1 and PC2), which also explain the most variance, were used to visualize the data.
\textbf{Figure~\ref{fig:fig3.1}b} shows the development and aging datasets projected on the PC1 and PC2 axis, where
the first principal component separates development and aging datasets, suggesting that expression changes may display different patterns in development and aging.
By calculating the median euclidian distance, I found that the median distance among development datasets is 21, 
whereas the median distance among aging datasets is 77. 
The fact that development datasets are clustered more closely compared to aging datasets may reflect an increase in heterogeneity during the aging period.

Then, the number of genes showing significant age-related gene expression change was calculated for development and aging datasets (\textbf{Figure~\ref{fig:fig3.1}c}).
The p-values obtained from linear regression analysis for each gene in each dataset were adjusted for multiple testing using the B{\&}H method (see \textbf{Section~\ref{subsec:p.adjust}}).
Analyzing significant changes, I first found that there are significantly more genes in the development datasets showing significant change compared to aging datasets (permutation test p-value = 0.003),
which may again reflect the higher heterogeneity during aging compared to development.
Second, the direction of change in development datasets is mainly in the positive direction (14 of 19 datasets showed more increase), 
whereas genes showing significant change during aging tend to decrease in expression (13 of 19 datasets showed more decrease).

Combined, my analysis of age-related expression changes demonstrated that the age-related gene expression changes may be less coordinated during aging,
leading to a higher inter-individual variability during aging, which may be the first clue to increased age-related heterogeneity during aging.

\section{Age-related expression heterogeneity change}
I next characterized age-related change in gene expression heterogeneity by performing Spearman's correlation test between absolute value of residuals and fourth root of age (see \textbf{Section~\ref{sec:het-change}}).
Similar to the previous section, I first examined the correlations in gene expression heterogeneity change among development and aging datasets (\textbf{Figure~\ref{fig:fig3.2}a}).
By conducting  Spearman's correlation test between all possible pairs of datasets among rho values, 
I found that aging datasets show higher correlation (median $\rho$ = 0.21, permutation test p-value = 0.24) in heterogeneity change
compared to development datasets (median $\rho$ = 0.11, permutation test p-value = 0.25), although both were not significant.
Whilst the difference in correlation of heterogeneity change between development and aging datasets was not significant (permutation test p-value = 0.2),
this observation may suggest that aging datasets show more similar changes in heterogeneity compared to development datasets.

\begin{figure}[h]
    \centering
    \includegraphics[width=.9\textwidth]{figures/figure3_2.png}
    \caption{Age-related heterogeneity change. 
    (a) The heatmap shows Spearman's correlation coefficient values for each pair of datasets.
    The row and column annotations reflect the brain region, data source and time period of the corresponding dataset.
    The datasets of each period were clustered separately.
    (b) Principal component analysis of age-related heterogeneity changes. Each point shows a dataset, 
    where color-coding reflects time period and point-shape shows the data source. 
    The values in parenthesis (on the axis labels) reflect the explained variance by the corresponding dataset.
    The values shown on the plot show the median euclidian distance among development and aging datasets, 
    calculated using PC1 and PC2.
    (c) Barplots show the number of genes showing significant age-related heterogeneity change (FDR p-value < 0.05) during development (left side) and aging (right side).
    Color-coding reflects the direction of heterogeneity change, which is determined by the sign of the rho value.
    Adapted from Isildak et al., 2020
    }~\label{fig:fig3.2}
\end{figure}

Principal component analysis of heterogeneity changes (i.e., rho values) was performed using a similar procedure as the previous section, 
except rho values were used instead of beta values. 
Similar to the PCA of expression changes, PCA of heterogeneity changes showed that development and aging datasets can be differentiated from each other by heterogeneity change (\textbf{Figure~\ref{fig:fig3.2}b}).
However, unlike expression change, development datasets tended to cluster more sparsely (median euclidian distance = 44), compared to aging datasets (median euclidian distance = 41).

I then investigated the genes showing significant changes in expression heterogeneity. 
The p-values obtained from Spearman's correlation test for each gene for each dataset were corrected for multiple testing using the B{\&}H method (see \textbf{Section~\ref{sec:het-change}}).
(\textbf{Figure~\ref{fig:fig3.2}c}) shows the number of genes showing significant heterogeneity change during development and aging.
First, I found that there is a small number of genes showing significant change during development compared to aging (permutation test p-value < 0.05).
Moreover, genes showing significant heterogeneity change during development tended to decrease in heterogeneity, whereas they showed a tendency to increase in aging datasets.
It is also important to note the high number of genes showing a significant change in Colantuoni2011, which is a result of the large sample size leading to increased statistical power.

Overall, this analysis demonstrated that there may be a higher consistency in heterogeneity change during aging compared to development, 
suggesting that increased heterogeneity may be a characteristic of the aging period.

\section{Consistent increase in heterogeneity during aging}
Having observed both that heterogeneity changes can differentiate between time periods and that there is a tendency towards more similar changes in heterogeneity during aging,
I next focused on identifying those genes showing the consistent change in heterogeneity.
As previously stated, in this study, I chose to focus on genes showing consistent trends among different datasets, rather than focusing on significant changes within individual datasets.
The most important advantage of this approach is that, using the advantage of having multiple datasets, 
it enables us to detect genes that would otherwise not pass the significance threshold due to low sample size.

\begin{figure}[h]
    \centering
    \includegraphics[width=.9\textwidth]{figures/figure3_3.png}
    \caption{(a) Boxplots on the left side shows the distribution of rho values, 
    whereas the barplots on the right shows the difference in medians between development and aging datasets.
    (b) Distribution of Spearman's correlation coefficients between expression and heterogeneity change of each dataset. 
    The color-coding reflects the data source, whereas 
    (c) Barplots showing expected and observed number of genes showing a consistent increase in heterogeneity among datasets during development (upper panel) and aging (lower panel).
    Adapted from Isildak et al., 2020
    }~\label{fig:fig3.3}
\end{figure}

I first analyzed the distribution of heterogeneity changes (i.e., $\rho$ values) for each dataset and time period (\textbf{Figure~\ref{fig:fig3.3}a}).
The 18 of 19 aging datasets showed more increase in age-related heterogeneity (median $\rho$ > 0), 
while the remaining one dataset showed no change in heterogeneity (median $\rho$ = 0). 
During development, on the other hand, I found that 14 of 19 datasets displayed a decrease in heterogeneity  (median $\rho$ < 0).
Moreover, a comparison of median heterogeneity changes between development and aging datasets revealed that 
18 of 19 datasets showed more increase during aging compared to development. 
The permutation test further demonstrated that the overall increase in heterogeneity during aging compared to development is significant (permutation test p-value < 0.001).

Although I observed increased heterogeneity during aging compared to development, 
one possible explanation may be related to the positive correlation between mean and variation, 
where increased heterogeneity may be a result of increased expression levels during aging, rather than being a biological effect.
To test this, I analyzed the correlations between expression and heterogeneity changes for each dataset and time period (\textbf{Figure~\ref{fig:fig3.3}b}).
Spearman's correlation test revealed that there is no significant dependence between expression and heterogeneity change.
In fact, development datasets showed higher positive correlations compared to aging datasets. 
Nevertheless, this analysis showed that the observed overall increase in heterogeneity during aging is not a result of mean-variance dependence.

Given that we observed an overall increase in heterogeneity during aging compared to development (\textbf{Figure~\ref{fig:fig3.3}a}),
and this increase is not a technical artifact (\textbf{Figure~\ref{fig:fig3.3}b}), I next focus on detecting those genes showing a consistent change in heterogeneity. 
For each gene, I calculated the number of datasets in which they show a  consistent increase in heterogeneity (in development and aging periods separately), 
irrespective of whether the increase is significant (\textbf{Figure~\ref{fig:fig3.3}c}).
The expected consistency in heterogeneity change was calculated using the random permutations (see \textbf{Section~\ref{subsec:perm.consist}} for details).
By comparing observed consistency in heterogeneity increase to expected consistency, I first found a shift towards increased heterogeneity consistency
during the aging period, whereas no such shift was observed in development datasets.
Moreover, 147 genes were identified that consistently increase in heterogeneity among all 19 aging datasets (permutation test p-value < 0.001), 
whereas there was only 1 gene showing a consistent decrease in all datasets during the aging period.
The full list of these consistent 147 is given in the \textbf{Appendix~\ref{app:const.genes}}.
However, it is also important to note that, according to permutations, 
the number of genes that would show consistent increase in heterogeneity under the null hypothesis (i.e., by chance) is 84,
indicating a 40\% true positives.
Although I couldn't confidently identify a gene set showing increased heterogeneity consistently among all datasets,
this analysis demonstrated a clear shift towards increased heterogeneity during aging compared to development. 

\section{Clustering}
I next examined if 147 consistent genes (i.e., showing consistent heterogeneity increase in all 19 datasets) display certain trajectories of heterogeneity change.
To test this, I grouped these consistent genes according to their heterogeneity change pattern using k-means clustering algoritm into 8 clusters (\textbf{Figure~\ref{fig:fig3.4}}).
Overall, the clustering analysis revealed three diverse patterns of heterogeneity change during aging:

\begin{enumerate}
    \item A steady increase in heterogeneity throughout the aging period (genes in clusters 3 and 7).
    \item A steady increase until the age of 60 years, followed by a slight fall in heterogeneity (genes in clusters 4, 5 and 8).
    \item A sharp increase in heterogeneity around the age of 60 years (genes in clusters 1, 2 and 6).
\end{enumerate}


\begin{figure}[h]
    \centering
    \includegraphics[width=.9\textwidth]{figures/figure3_4.png}
    \caption{Trajectories of different clusters of 147 genes showing consistent increase in all 19 datasets during aging.
    The x-axis shows age in years, while the y-axis shows scaled heterogeneity changes (i.e., rho values).
    The spline lines show the mean heterogeneity change level for genes. 
    Adapted from Isildak et al., 2020
    }~\label{fig:fig3.4}
\end{figure}

\section{Functional analysis}
To investigate the functional role of increased heterogeneity during aging, 
I next performed gene set enrichment analysis using Gene Ontology (GO) Biological Process categories~\cite{GO2019} and Kyoto Encyclopedia of Genes and Genomes (KEGG) pathways~\cite{Kanehisa2019}.
The gene set enrichment analysis was performed on the genes that were ranked by number of datasets they show consistent increase in heterogeneity. 

\subsection{GO Biological Process enrichment analysis}
Gene Ontology Biological Process enrichment analysis was performed on development and aging datasets, separately. 
While there was no GO term significantly enriched for the consistent changes in heterogeneity during development, 
I identified 111 significantly enriched GO Biological Process terms for the consistent changes in aging. 
The full list of enriched GO terms and normalized enrichment scores can be found in \textbf{Table~\ref{table:a2}}, 
while \textbf{Figure~\ref{fig:fig3.5}} shows the REVIGO summarization.

One of the most important enriched biological process GO term was the autophagy (GO term ID: GO:0006914), which was previously suggested to play important role in aging and aging-related diseases~\cite{Luo2019}.
Moreover, another important enriched group of genes belonged to the axon regeneration (GO term ID: GO:0048679), which also demonstrated to decrease during aging~\cite{Belin2014}.
Additionally, the regulation of T-helper cell differentiation and cellular response to virus terms were found to be significantly enriched. 
Given the weakened immune system in aging, this result also suggests that increased heterogeneity may have important consequences in aging.
Other significantly enriched groups included terms related to metabolic processes, mRNA processing and localization.
Overall, GO Biological Process enrichment analysis of genes showing consistent increaese in hetergeneity indicates that increased heterogeneity may be associated with the aging-related phenotypes.

\begin{landscape}
\begin{figure}[h]
    \centering
    \includegraphics[width=1.5\textwidth]{figures/figure3_5.png}
    \caption{GO Biological Process enrichment analysis results for the genes showing consistent increase in heterogeneity, summarized by REVIGO~\cite{Supek2011}.
    }~\label{fig:fig3.5}
\end{figure}
\end{landscape}

\subsection{KEGG pathway enrichment analysis}
Next, I performed KEGG pathway enrichment analysis. Similarly, there was no KEGG pathway found to be significantly enriched in development period.
In aging, on the other hand, 

\begin{figure}[h]
    \centering
    \includegraphics[width=.9\textwidth]{figures/figure3_6.png}
    \caption{KEGG pathways enrichment analysis results for the genes showing consistent increase in heterogeneity.
    (a) Significantly enriched KEGG pathways (on the y-axis) and the distribution of number of datasets in which genes showing consistent increase in heterogeneity (x-axis). 
    (b) Demonstration of Longevity Regulation Pathway as an example during development (upper panel) and aging (lower panel). Nodes represent genes, while color-coding reflect the consistency in heterogeneity increase.
    Adapted from Isildak et al., 2020
    }~\label{fig:fig3.6}
\end{figure}

\section{Transcriptional regulation enrichment analysis}

\begin{figure}[h]
    \centering
    \includegraphics[width=.9\textwidth]{figures/figure3_7.png}
    \caption{The distribution of correlation coefficients between number of transcriptional regulators and heterogeneity changes. 
    Adapted from Isildak et al., 2020
    }~\label{fig:fig3.7}
\end{figure}

\subsection{Transcription factor enrichment analysis}

\subsection{miRNA enrichment analysis}